src/node_contextify.cc

```diff
@@ -240,30 +256,39 @@ MaybeLocal<Context> ContextifyContext::CreateV8Context(
   ctx->SetEmbedderData(ContextEmbedderIndex::kAllowWasmCodeGeneration,
                        options.allow_code_gen_wasm);
 
+  Utf8Value name_val(env->isolate(), options.name);
   ContextInfo info(*name_val);
-
   if (!options.origin.IsEmpty()) {
     Utf8Value origin_val(env->isolate(), options.origin);
     info.origin = *origin_val;
   }
 
+  {
+    Context::Scope context_scope(ctx);
+    Local<String> ctor_name = sandbox_obj->GetConstructorName();
+    if (!ctor_name->Equals(ctx, env->object_string()).FromMaybe(false) &&
+        ctx->Global()
+            ->DefineOwnProperty(
+                ctx,
+                v8::Symbol::GetToStringTag(env->isolate()),
+                ctor_name,
+                static_cast<v8::PropertyAttribute>(v8::DontEnum))
+            .IsNothing()) {
+      return false;
+    }
+  }
+
   env->AssignToContext(ctx, info);
 
-  return scope.Escape(ctx);
+  // This should only be done after the initial initializations of the context
+  // global object is finished.
+  ctx->SetAlignedPointerInEmbedderData(ContextEmbedderIndex::kContextifyContext,
```

Comment:
Do we still need this as `ContextEmbedderIndex::kContextifyContext` already initialized in `AssignToContext`?

Reply:
nvm. I overlooked the second parameter.